Device Data

Krishnan-Priyanshu opened the pull request:
Hi Team,

Please review the PR and will be updating the go sum after the goscaleio pr merge

![image](https://github.com/dell/terraform-provider-powerflex/assets/118425422/3f2ad6cb-6a8b-44bb-96f8-0291879556c6)

shenda1 commented on examples/data-sources/powerflex_device/data-source.tf:
```diff
@@ -0,0 +1,9 @@
+
+data "powerflex_device" "dev" {
+}
+
```
can we add examples to get devices based on storage pool and sds?

Krishnan-Priyanshu replied:
add the example to the config file.

shenda1 commented on powerflex/device_datasource_schema.go:
```diff
@@ -0,0 +1,583 @@
+				stringvalidator.ConflictsWith(path.MatchRoot("storage_pool_name")),
+			},
+		},
+		"current_path": schema.StringAttribute{
```
Are we filtering based on the path as well?

shenda1 commented on powerflex/device_datasource.go:
```diff
@@ -0,0 +1,431 @@
+			state.StoragePoolID = types.StringValue(sp.ID)
+		}
+
+		if !state.StoragePoolID.IsUnknown() {
```
Better to have else if as storage pool id and name are mutually exclusive

Krishnan-Priyanshu replied:
added the else if block for mutual exclusive attributes

shenda1 commented on powerflex/device_datasource_schema.go:
```diff
@@ -0,0 +1,583 @@
+func getAccelerationPropsParamsSchema() map[string]schema.Attribute {
+	return map[string]schema.Attribute{
+		"acc_used_capacity_in_kb": schema.StringAttribute{
+			Description:         "Acc Used Capacity In KB Acceleration Properties Parameters Of The Device Instance.",
```
Please verify the description once. Cant relate whats acc here.

Krishnan-Priyanshu replied:
Accelearation Used capacity in kb refers to the amount of data that has been utilized or consumed by an acceleration system or process in kb.
By monitoring the acceleration used capacity in KB, one can assess the efficiency, effectiveness, or utilization of the acceleration system.

Krishnan-Priyanshu replied:
I've updated the description to acceleration

Krishnan-Priyanshu replied:
do you mean to change the config name?
as : acceleration_used_capacity_in_kb

shenda1 replied:
Acceleration Used capacity is fine.

shenda1 commented on powerflex/device_datasource_schema.go:
```diff
@@ -0,0 +1,583 @@
+		Computed:            true,
+	},
+	"led_setting": schema.StringAttribute{
+		Description:         "Led Setting Of The Device Instance.",
```
LED?

Krishnan-Priyanshu replied:
updated to LED

shenda1 commented on powerflex/device_datasource_schema.go:
```diff
@@ -0,0 +1,583 @@
+			Computed:            true,
+		},
+		"dest_checksum_changelog_acc_device_id": schema.StringAttribute{
+			Description:         "Dest Checksum Changelog Acc Device ID Storage Properties Parameters Of The Device Instance.",
```
Dest is Destination?

Krishnan-Priyanshu replied:
Updated to Destination

AnikaAgiwal2711 commented on powerflex/device_datasource_schema.go:
```diff
@@ -0,0 +1,583 @@
+		Computed:            true,
+	},
+	"device_state": schema.StringAttribute{
+		Description:         "List of devices fetched Of The Device Instance.",
```
I believe this should be state of the device instance

Krishnan-Priyanshu replied:
updated the description

rounak-adhikary commented on powerflex/device_datasource.go:
```diff
@@ -0,0 +1,428 @@
+			}
+		}
+	} else if !state.SdsID.IsNull() || !state.SdsName.IsNull() {
+		if !state.SdsName.IsNull() {
```
I do not believe that we are fetching device by sds name (its probably always giving an empty list of devices).
I see that the test is not checking the state generated, so we are probably missing this scenario.
```
{
	Config: ProviderConfigForTesting + deviceDataWithSdsName,
},
```
Can u pls verify?
Can u pls update all the tests so that we check for the state?

rounak-adhikary commented on powerflex/device_datasource.go:
```diff
@@ -0,0 +1,428 @@
+				)
+				return
+			}
+			sp, err1 := pd.FindStoragePool("", state.StoragePoolName.ValueString(), "")
```
Same issue with storage pool name... we are never fetching all the devices from that storage pool...

Krishnan-Priyanshu replied:
fixed the issue

Krishnan-Priyanshu commented:
coverage report after addressing the review comments
![image](https://github.com/dell/terraform-provider-powerflex/assets/118425422/0960976f-6388-4c99-a968-cac3dc594c7f)

rounak-adhikary reviewed (approved):
Not a fan of hardcoded testcases, but all else looks good.

Krishnan-Priyanshu commented:
> Not a fan of hardcoded testcases, but all else looks good.

No issue, I will create new PR to fix the hardcoded testcase 😃
